crates/storage/provider/src/providers/database/mod.rs

```diff
@@ -663,46 +672,17 @@ mod tests {
         );
 
         // Checkpoint and no gap
-        provider
-            .tx_ref()
-            .put::<tables::CanonicalHeaders>(head.number, head.hash())
-            .expect("failed to write canonical");
-        provider
-            .tx_ref()
-            .put::<tables::Headers>(head.number, head.clone().unseal())
-            .expect("failed to write header");
+        let mut snapshot_writer = provider
+            .snapshot_provider()
+            .expect("should exist")
+            .latest_writer(SnapshotSegment::Headers)
+            .unwrap();
+        snapshot_writer.append_header(head.header.clone(), U256::ZERO, head.hash()).unwrap();
+        snapshot_writer.commit().unwrap();
+        drop(snapshot_writer);
 
         let gap = provider.sync_gap(mode.clone(), checkpoint).unwrap();
         assert_eq!(gap.local_head, head);
         assert_eq!(gap.target.tip(), consensus_tip.into());
-
```

Comment:
with static files, there's no long the possibility of having a gap where the tip is already in the database